common/src/main/java/com/radixdlt/crypto/ECDSASecp256k1PublicKey.java

```diff
@@ -66,107 +66,108 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
-import com.google.common.base.Suppliers;
 import com.google.common.hash.HashCode;
 import com.radixdlt.crypto.exception.PublicKeyException;
-import com.radixdlt.lang.Option;
 import com.radixdlt.sbor.codec.CodecMap;
 import com.radixdlt.sbor.codec.CustomByteArrayCodec;
 import com.radixdlt.serialization.DsonOutput;
 import com.radixdlt.utils.Bytes;
 import java.util.Arrays;
-import java.util.Objects;
 import java.util.Optional;
-import java.util.function.Supplier;
 import org.bouncycastle.math.ec.ECPoint;
 
-/** Asymmetric EC public key provider fixed to curve 'secp256k1' */
+/**
+ * A wrapper for EC (fixed to curve 'secp256k1') public key bytes in a compressed format. Note that
+ * this class does NOT verify that the bytes represent a valid point on a curve (except in
+ * `decodePoint()` method).
+ */
 public final class ECDSASecp256k1PublicKey {
   public static void registerCodec(CodecMap codecMap) {
     codecMap.register(
         ECDSASecp256k1PublicKey.class,
         codecs ->
             new CustomByteArrayCodec<>(
-                ECDSASecp256k1PublicKey::getCompressedBytes,
-                ECDSASecp256k1PublicKey::fromCompressedBytesUnchecked));
+                ECDSASecp256k1PublicKey::getBytes,
+                compressedBytes -> {
+                  try {
+                    return ECDSASecp256k1PublicKey.fromBytes(compressedBytes);
+                  } catch (PublicKeyException e) {
+                    throw new RuntimeException("ECDSASecp256k1PublicKey couldn't be decoded", e);
```

Comment:
> Would it be more convenient to make it extends RuntimeException?

Hmh, maybe a bit more convenient ...but I'd then prefer `Result<PubKey, Err>` instead of a runtime exception (to make sure it's always handled). I think using a checked exception is justified here.

> Should it be 2 different exceptions?

We could use separate exceptions, but I don't see a strong reason why that's needed?